aiplatform/apiv1beta1/deployment_resource_pool_client.go

```diff
@@ -0,0 +1,672 @@
+	iampb "google.golang.org/genproto/googleapis/iam/v1"
+	longrunningpb "google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/proto"
+)
+
+var newDeploymentResourcePoolClientHook clientHook
+
+// DeploymentResourcePoolCallOptions contains the retry settings for each method of DeploymentResourcePoolClient.
+type DeploymentResourcePoolCallOptions struct {
+	CreateDeploymentResourcePool []gax.CallOption
+	GetDeploymentResourcePool    []gax.CallOption
+	ListDeploymentResourcePools  []gax.CallOption
+	DeleteDeploymentResourcePool []gax.CallOption
+	QueryDeployedModels          []gax.CallOption
+	GetLocation                  []gax.CallOption
+	ListLocations                []gax.CallOption
+	GetIamPolicy                 []gax.CallOption
+	SetIamPolicy                 []gax.CallOption
+	TestIamPermissions           []gax.CallOption
+	CancelOperation              []gax.CallOption
+	DeleteOperation              []gax.CallOption
+	GetOperation                 []gax.CallOption
+	ListOperations               []gax.CallOption
+	WaitOperation                []gax.CallOption
+}
+
+func defaultDeploymentResourcePoolGRPCClientOptions() []option.ClientOption {
+	return []option.ClientOption{
+		internaloption.WithDefaultEndpoint("aiplatform.googleapis.com:443"),
+		internaloption.WithDefaultMTLSEndpoint("aiplatform.mtls.googleapis.com:443"),
+		internaloption.WithDefaultAudience("https://aiplatform.googleapis.com/"),
+		internaloption.WithDefaultScopes(DefaultAuthScopes()...),
+		internaloption.EnableJwtWithScope(),
+		option.WithGRPCDialOption(grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(math.MaxInt32))),
+	}
+}
+
+func defaultDeploymentResourcePoolCallOptions() *DeploymentResourcePoolCallOptions {
+	return &DeploymentResourcePoolCallOptions{
+		CreateDeploymentResourcePool: []gax.CallOption{},
+		GetDeploymentResourcePool:    []gax.CallOption{},
+		ListDeploymentResourcePools:  []gax.CallOption{},
+		DeleteDeploymentResourcePool: []gax.CallOption{},
+		QueryDeployedModels:          []gax.CallOption{},
+		GetLocation:                  []gax.CallOption{},
+		ListLocations:                []gax.CallOption{},
+		GetIamPolicy:                 []gax.CallOption{},
+		SetIamPolicy:                 []gax.CallOption{},
+		TestIamPermissions:           []gax.CallOption{},
+		CancelOperation:              []gax.CallOption{},
+		DeleteOperation:              []gax.CallOption{},
+		GetOperation:                 []gax.CallOption{},
+		ListOperations:               []gax.CallOption{},
+		WaitOperation:                []gax.CallOption{},
+	}
+}
+
+// internalDeploymentResourcePoolClient is an interface that defines the methods available from Vertex AI API.
+type internalDeploymentResourcePoolClient interface {
+	Close() error
+	setGoogleClientInfo(...string)
+	Connection() *grpc.ClientConn
+	CreateDeploymentResourcePoolOperation(name string) *CreateDeploymentResourcePoolOperation
```

Comment:
Where is the `CreateDeploymentResourcePool` method in this client? I don't see it in the test either.

I'm trying to find the impl used in this snippet:

```
c, err := aiplatform.NewDeploymentResourcePoolClient(ctx)
	if err != nil {
		// TODO: Handle error.
	}
	defer c.Close()

	req := &aiplatformpb.CreateDeploymentResourcePoolRequest{
		// TODO: Fill request struct fields.
		// See https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/aiplatform/v1beta1#CreateDeploymentResourcePoolRequest.
	}
	op, err := c.CreateDeploymentResourcePool(ctx, req)
```

Reply:
There is none. aiplatform does not yet REST transports. It is likely there is a diff between last regen and this PR that caused a diff of a new feature to be added here. But this is unrelated to transport stuff.